check_all_options.c

```diff
@@ -101,6 +101,30 @@ void check_options_tcp_rr(struct options *opts, struct callbacks *cb)
 
 void check_options_tcp_stream(struct options *opts, struct callbacks *cb)
 {
+#ifdef WITH_TCPDIRECT
+      if (opts->tcpd_gpu_pci_addr) {
+            CHECK(cb, opts->tcpd_nic_pci_addr,
+                  "Must provide NIC PCI address if GPU PCI address was provided.");
+            CHECK(cb, opts->tcpdirect_phys_len > 0,
+                  "Must provide non-zero --tcpdirect-phys-len flag if GPU PCI address was provided.");
+            // TODO check page-alignment
+            // CHECK((CUdeviceptr)gpu_gen_mem_ % PAGE_SIZE == 0);
+
+            if (opts->client) {
+                  CHECK(cb, !opts->tcpd_rx_cpy,
+                        "Copying CUDA buffer to userspace only allowed on hosts.");
+            } else {
+                  CHECK(cb, opts->tcpdirect_src_ip,
+                        "Must provide source IP address for TCPDirect host.");
+                  CHECK(cb, opts->tcpdirect_dst_ip,
+                        "Must provide destination IP address for TCPDirect host.");
+            }
+            CHECK(cb, opts->num_flows == opts->num_threads,
+                  "Thread/Flow count must be equal when running in TCPDirect mode.");
```

Comment:
Why not fall back to neper's default number of thread/flow count? Just a suggestion, not a high priority.

Reply:
buffers:threads are 1:1, but we want buffers:flows to be 1:1 as well. This is a TCP devmem specific restriction/hack